exo-go/src/types/dependency.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import "os"
+
+// Dependency represents a dependency of an application
+type Dependency struct {
+	Name    string
+	Version string
+	Silent  bool             `yaml:",omitempty"`
+	Config  DependencyConfig `yaml:",omitempty"`
+}
+
+// DependencyConfig represents the configuration of a dependency
+type DependencyConfig struct {
```

Comment:
I would like to suggest putting each struct into its own file, and matching the file name with the struct name. That's a best practice from Ruby that is incredibly useful.